src/asl.ts

```diff
@@ -696,9 +736,9 @@ export class ASL {
             assign: isForOfStmt(stmt)
               ? {
                   Type: "Pass",
-                  node: stmt.variableDecl,
+                  node: stmt.initializer,
                   InputPath: `${tempArrayPath}[0]`,
-                  ResultPath: `$.${stmt.variableDecl.name}`,
+                  ResultPath: `$.${stmt.initializer.getName()}`,
```

Comment:
Why can't we be type safe here? This will be nasty when we go to add BindingElm and need to check the type anyways (soon).

```ts
if(isVariableDecl(stmt.initializer) && isBindingElm(stmt.initializer.name)) {
    throw...
}
const initializerName = stmt.initializer.name.name; 
```

Reply:
Also I realize I wasn't checking this before in a typesafe way, good argument for name being an identifier... and disappointing....